metadata/src/test/java/org/apache/kafka/metalog/LocalLogManager.java

```diff
@@ -723,9 +727,35 @@ public long scheduleAtomicAppend(int epoch, List<ApiMessageAndVersion> batch) {
 
     @Override
     public void resign(int epoch) {
-        LeaderAndEpoch curLeader = leader;
-        LeaderAndEpoch nextLeader = new LeaderAndEpoch(OptionalInt.empty(), curLeader.epoch() + 1);
-        shared.tryAppend(nodeId, curLeader.epoch(), new LeaderChangeBatch(nextLeader));
+        if (epoch < 0) {
+            throw new IllegalArgumentException("Attempt to resign from an invalid negative epoch " + epoch);
+        }
+
+        LeaderAndEpoch leaderAndEpoch = leaderAndEpoch();
+        int currentEpoch = leaderAndEpoch.epoch();
+
+        if (epoch > currentEpoch) {
+            throw new IllegalArgumentException("Attempt to resign from epoch " + epoch +
+                    " which is larger than the current epoch " + currentEpoch);
```

Comment:
The contract in `RaftClient` still says 

```java
     * @param epoch the epoch to resign from. If this does not match the current epoch, this
     *              call will be ignored.
```

Of course `KafkaRaftClient` is doing the same check, so I guess we need to update the `RaftClient` javadoc.

Reply:
Thanks for catching. I have fixed it as per your suggestion in the latest commit.